parlai/tasks/eli5/agents.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from parlai.core.teachers import FixedDialogTeacher
+from .build import build
+import os
+import json
+
+
+class ELI5Teacher(FixedDialogTeacher):
+    """
+    ELI5 Teacher, taken from https://github.com/facebookresearch/ELI5.
+    """
+
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+        # self.add_cmdline_args(opt)
+        build(opt)
+        # self._setup_data(opt['datafile'], jsons_path)
+        self.id = 'eli5'
+        self.messages = self.load_eli5(opt)
+        self.reset()
+
+    @staticmethod
+    def add_cmdline_args(argparser):
+        group = argparser.add_argument_group('ELI5 Knowledge arguments')
+        group.add_argument(
+            '--knowledge',
+            type='bool',
+            default=True,
+            help='Whether to include supporting document knowledge',
+        )
+
+    def load_eli5(self, opt):
+        """
+        Load data based on data split.
+        """
+        dp = opt['datapath']
+        dt = opt['datatype'].split(':')[0]
+        eli_path = "eli5/processed_data/selected_15_1/explainlikeimfive_"
+        fname = os.path.join(dp, eli_path + dt + ".json")
+        opt['datafile'] = fname
+        with open(fname) as json_file:
+            data = json.load(json_file)
+        ds = []
+        for d in data:
+            if bool(self.opt['knowledge']) and self.opt['knowledge'] != 'False':
```

Comment:
nit: you don't need the `bool()` check here, having an arg type of `'bool'` auto-parses `False,false,0` and `True,true,1` to be `False` and `True` respectively